File: lcci/10.02.py
from typing import List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = dict()
        for s in strs:
            if sorted(s) in d:
                d[sorted(s)].append(s)
            else:
                d[sorted(s)] = [s]
        res = list()
        for k, v in d.items():
            res.append(v)
        return res

File: lcci/10.02.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    vector<vector<string>> groupAnagrams(vector<string> &strs) {
        map<string, vector<string>> mp;
        for (auto str : strs) {
            string s(str);
            sort(s.begin(), s.end());
            if (mp.count(s)) { // 已经有了
                mp[s].push_back(str);
            } else {
                mp[s] = vector{str};
            }
        }
        vector<vector<string>> res;
        for (auto p : mp) { res.push_back(p.second); }
        return res;
    }
};

int main() {
    //
    return 0;
}